### PR TITLE
Fix blueprints status codes and add test

### PR DESCRIPTION
**Pull request by avitova**
We were missing a few 404 status codes. I added a few lines returning these, and also added some tests.

### CODE REVIEW

File: internal/v1/handler_blueprints_test.go [hunk 1]
```diff
@@ -295,6 +299,14 @@ func TestHandlers_GetBlueprintComposes(t *testing.T) {
...
 	require.Equal(t, blueprintId, *result.Data[0].BlueprintId)
 	require.Equal(t, 2, *result.Data[0].BlueprintVersion)
+
+	//// get composes for non-existing blueprint
+	respStatusCode, body = tutils.GetResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/experimental/blueprints/%s/composes?blueprint_version=1", uuid.New().String()), &tutils.AuthString0)
+	require.Equal(t, 200, respStatusCode)
+	err = json.Unmarshal([]byte(body), &result)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(result.Data))
+	require.Equal(t, 0, result.Meta.Count)
```
Comment (ezr-ondrej) [inline internal/v1/handler_blueprints_test.go:309]: I'm unsure here: should we rather return 404 as well, this feels bit weird to have empty response for non-existent blueprint

**Review by ezr-ondrej — COMMENTED** [review]
Nice improvements! :orange_heart:
Do you think we should also return 404 on listing composes for non-existent blueprint?

File: internal/v1/handler_blueprints_test.go (hunk 1 above)
Comment (avitova) [reply internal/v1/handler_blueprints_test.go:309]: Yes, this is interesting; we'd have to distinguish between Blueprint Not Found vs No Composes. This is the case for most of the endpoints where we use JOINS. I am happy to discuss how much we want to enhance this.

**Comment by avitova** [comment]
I have no strong opinion on this; I see a difference between 404 for a non-existent blueprint and 200 for empty composes for a given blueprint. I would have a more extensive discussion on the topic outside GitHub and address this in different PR. Does that sound good to you?

**Comment by ezr-ondrej** [comment]
> I would have a more extensive discussion on the topic outside GitHub and address this in different PR. Does that sound good to you?

Agreed, lets just not add tests for it here as what's tested is intentional :) and when we don't know how it should behave, lets not test it ;)

File: internal/v1/handler_blueprints_test.go (hunk 1 above)
Comment (lzap) [reply internal/v1/handler_blueprints_test.go:309]: If a bluprint does not exist in a `/blueprints/XYZ/whatever` path to me it feels logical to see Not Found just like this was a directory on a file system.

**Comment by ezr-ondrej** [comment]
Let me know once this is ready again, I still see the test there, so I'll hold off the review :)

**Comment by avitova** [comment]
The mentioned test is for: "existing blueprint has no composes" case. Imo that does not have to be 404, just an empty list, and a 200 response.
There is an additional test for listing compossd of non-existing blueprint => ends up with 404.

**Review by ezr-ondrej — APPROVED** [review]
Thanks! :)

**Comment by ezr-ondrej** [comment]
Oh wow, I was sure I've merged this, sorry! Merging.

**Comment by avitova** [comment]
Thank you!:)

**Pull request merged**